cmd/juju/action/common.go

```diff
@@ -146,15 +149,34 @@ func (c *runCommandBase) processOperationResults(ctx *cmd.Context, results *acti
 		}
 		return nil
 	}
-	return c.waitForTasks(ctx, tasks, info)
+	failed, err := c.waitForTasks(ctx, tasks, info)
+	if err != nil {
+		return errors.Trace(err)
+	} else if len(failed) > 0 {
+		var plural string
+		if len(failed) > 1 {
+			plural = "s"
+		}
+		list := make([]string, 0, len(failed))
+		for k, v := range failed {
+			list = append(list, fmt.Sprintf(" - id %q with return code %d", k, v))
```

Comment:
s/return/exit?

Also, it could be helpful to include the machine in the error, e.g `-id %q on machine %q with exit code %d`

Reply:
It isn't always a machine, so the error message get's very complicated.

Reply:
What else can it be? The task ID by itself isn't really useful as the user would presumably have to look it up somehow, right?

Reply:
Anything that uses names, but in reality it's probably machine or unit.

Reply:
The receiver of an action (ie where a task can run) is either a machine or unit.
The task ID is useful as it can be used with show-task.
Perhaps the error message could be enhanced to hint that `show-task` can be used to inspect the details on any failed task IDs